includes/Core/Modules/Module.php

```diff
@@ -861,6 +867,10 @@ protected function exception_to_error( Exception $e, $datapoint ) {
 			'reason' => $reason,
 		);
 
+		if ( $cache_ttl ) {
+			$data['cacheTTL'] = $cache_ttl;
+		}
+
```

Comment:
While this works, the entire PHP logic needed here is specific to the Analytics API, so I don't think it is suitable to be added to the general `Module` class (also see "the `Analytics` exception handling function" reference in the ACs). Let's instead put these changes into the `Analytics` class:
* Modify the overridden `exception_to_error` method, which already includes the necessary check.
* Instead of immediately returning the `WP_Error` returned by the parent method, modify it to add the `cacheTTL` to the `$data` property of the `WP_Error` object based on the checks made.
    * Basically, the existing clause can be moved below the parent method call so that we can then within it update the `WP_Error` as needed before returning it.

Reply:
Ah! Makes sense @felixarntz I thought we were going to do that for all "Restricted metrics..." error. The PR has been updated. Can you take a look @eugene-manuilov ?